llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7103,6 +7103,37 @@ Instruction *InstCombinerImpl::foldICmpCommutative(ICmpInst::Predicate Pred,
   if (Value *V = foldICmpWithLowBitMaskedVal(Pred, Op0, Op1, Q, *this))
     return replaceInstUsesWith(CxtI, V);
 
+  // Folding (X / Y) pred X => X ~pred 0 for some constant Y other than 0 or 1
+  {
+    const APInt *Divisor;
+    if (match(Op0, m_UDiv(m_Specific(Op1), m_APInt(Divisor))) &&
+        Divisor->ugt(1)) {
+      return new ICmpInst(ICmpInst::getSwappedPredicate(Pred), Op1,
+                          Constant::getNullValue(Op1->getType()));
+    }
+
+    if (match(Op0, m_SDiv(m_Specific(Op1), m_APInt(Divisor))) &&
+        Divisor->ugt(1) && !ICmpInst::isUnsigned(Pred)) {
```

Comment:
```suggestion
    if (!ICmpInst::isUnsigned(Pred) && match(Op0, m_SDiv(m_Specific(Op1), m_APInt(Divisor))) &&
        Divisor->ugt(1)) {
```
`!ICmpInst::isUnsigned(Pred)` is cheaper.

Reply:
Fixed.